contracts/VeListaAutoCompounder.sol

```diff
@@ -0,0 +1,270 @@
+        address _velista,
+        address _veListaDistributor,
+        address _oracle,
+        address _feeReceiver,
+        address _admin
+    ) public initializer {
+        require(
+            _lista != address(0) &&
+                _velista != address(0) &&
+                _veListaDistributor != address(0) &&
+                _feeReceiver != address(0) &&
+                _admin != address(0) &&
+                _oracle != address(0),
+            "Zero address provided"
+        );
+        __AccessControl_init();
+        _setupRole(DEFAULT_ADMIN_ROLE, _admin);
+
+        lista = IERC20(_lista);
+        veLista = IVeLista(_velista);
+        require(lista == veLista.token(), "Invalid Lista token");
+
+        veListaDistributor = IVeListaDistributor(_veListaDistributor);
+        require(veLista == veListaDistributor.veLista(), "Invalid veLista");
+
+        oracle = OracleInterface(_oracle);
+        feeReceiver = _feeReceiver;
+
+        feeRate = 300; // 3%
+        maxFeeRate = 1000; // 10%
+        require(
+            feeRate <= maxFeeRate && maxFeeRate <= 10000,
+            "Invalid fee rate"
+        );
+
+        autoCompoundThreshold = 5 * 1e18; // $5
+        maxFee = 10 * 1e18; // $10
+        require(
+            (feeRate * autoCompoundThreshold) / 10000 <= maxFee,
+            "Invalid fee"
+        );
+
+        emit VeListaDistributorUpdated(_veListaDistributor);
+        emit OracleUpdated(_oracle);
+    }
+
+    /**
+     * @dev Auto-compound the veLista rewards.
+     *      Claim LISTA from veListaDistributor and then compound LISTA by doing `increaseAmountFor`.
+     *
+     */
+    function claimAndIncreaseAmount(uint16 toWeek) public {
+        address account = msg.sender;
+        require(autoCompoundEnabled[account], "Auto compound not enabled");
+
+        IVeLista.AccountData memory data = veLista.getLockedData(account);
+        require(data.locked > 0, "No locked amount");
+
+        (uint256 claimableAmt, ) = veListaDistributor.getTokenClaimable(
+            account,
+            address(lista),
+            toWeek
+        );
+        require(
+            claimableAmt > 0 && isEligibleForAutoCompound(claimableAmt),
+            "Not eligible for auto compound"
+        );
+
+        uint256 claimedAmount = veListaDistributor.claimForCompound(
+            account,
+            address(lista),
+            toWeek
+        );
+        uint256 amtToCompound = getAmountToCompound(claimedAmount);
+
+        IVeLista(veLista).increaseAmountForCompound(account, amtToCompound);
+
+        totalFee += claimedAmount - amtToCompound;
+
+        emit AutoCompounded(account, address(lista), claimedAmount, amtToCompound);
+    }
+
+    function withdrawFee() external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(totalFee > 0, "No fee to withdraw");
+
+        uint256 fee = totalFee;
+        totalFee = 0;
+
+        lista.safeTransfer(feeReceiver, fee);
+        emit feeWithdrawn(feeReceiver, fee);
```

Comment:
`feeWithdrawn` To `FeeWiithdrawn`

Reply:
done in 13945e308d471b9045e67d3d07c56caa7e1c1455